controllers/cluster_controller.go

```diff
@@ -267,12 +270,18 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 	// AddFinalizer	is idempotent
 	controllerutil.AddFinalizer(cluster, ClusterFinalizerName)
 
-	if cluster.Spec.BundlesRef == nil {
+	if cluster.Spec.BundlesRef == nil && cluster.Spec.EksaVersion == nil {
 		if err = r.setBundlesRef(ctx, cluster); err != nil {
 			return ctrl.Result{}, err
 		}
 	}
 
+	if cluster.Spec.EksaVersion == nil && cluster.Spec.BundlesRef == nil {
+		if err = r.setDefaultEksaVersion(ctx, cluster); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
```

Comment:
I think the way this logic operates is wrong
Also we might benefit from combining these two methods so the code looks more coherent
Let's sync offline, it will be easier